File: problems/algorithms/1400/1400_k_pal.py
class Solution:
    def canConstruct(self, s: str, k: int) -> bool:
        if len(s) < k:
            return False

        letters = [0] * 26

        for letter in s:
            letters[ord('a') - ord(letter)] += 1
        
        odd = 0
        for val in letters:
            if val % 2 == 1:
                odd += 1
        
        return odd <= k

File: problems/algorithms/1400/1400_k_pal.c
#include <stdbool.h>

bool canConstruct(char* s, int k) {
    int size = 0;
    int arr[26] = {0};

    while (*s != '\0') {
        arr[*s - 'a'] ++;
        s++;
        size ++;
    }

    if (size < k){
        return false;
    }

    int odd = 0;
    for(int i=0; i<26; i++){
        if (arr[i] % 2 == 1){
            odd ++;
        }
    }
    
    return odd <= k;
}